lib/arduino_ci/cpp_library.rb

```diff
@@ -410,9 +410,9 @@ def all_arduino_library_dependencies!(additional_libraries = [])
       recursive = (additional_libraries + arduino_library_dependencies).map do |n|
         other_lib = self.class.new(n, @backend)
         other_lib.install unless other_lib.installed?
-        other_lib.all_arduino_library_dependencies!
+        [n] + other_lib.all_arduino_library_dependencies!
       end.flatten
-      (additional_libraries + recursive).uniq
+      recursive.uniq
     end
```

Comment:
Do you have a test case or some performance metric that relates to this change?  Just trying to figure out whether this is an improved implementation or a bugfix